chia/wallet/wallet_state_manager.py

```diff
@@ -1430,6 +1431,25 @@ async def get_all_transactions(self, wallet_id: int) -> List[TransactionRecord]:
     async def get_transaction(self, tx_id: bytes32) -> Optional[TransactionRecord]:
         return await self.tx_store.get_transaction_record(tx_id)
 
+    async def get_transaction_by_wallet_record(self, wr: WalletCoinRecord) -> Optional[TransactionRecord]:
+        records = await self.tx_store.get_transactions_by_height(wr.confirmed_block_height)
+        for record in records:
+            if wr.coin in record.additions:
```

Comment:
This whole function is heuristic so it seems a little icky on the whole, but why just additions?  Wouldn't a coin being in the removals make the transaction just as relevant to that coin record?

Reply:
sure